File: app/src/main/java/liyihuan/app/android/androidpractice/http/BaseRepository.java
package liyihuan.app.android.androidpractice.http;

import liyihuan.app.android.androidpractice.datasource.Utils;

/**
 * @ClassName: BaseRepository
 * @Description: java类作用描述
 * @Author: liyihuan
 * @Date: 2021/4/21 21:55
 */
public abstract class BaseRepository<T> {

    protected T apiService;

    public BaseRepository() {
        apiService = Renovace.create((Class<? extends T>) Utils.findNeedType(getClass()));
    }

}

File: app/src/main/java/liyihuan/app/android/androidpractice/http2/BaseRepository.kt
package liyihuan.app.android.androidpractice.http2

import liyihuan.app.android.androidpractice.http2.request.HttpProvider


/**
 * @author created by liyihuanx
 * @date 2021/9/7
 * @description: 类的描述
 */
abstract class BaseRepository<T> {

    var apiService: T = HttpProvider.defaultCreate(TypeUtils.findNeedType(javaClass) as Class<out T>)

}